CRM/Utils/System/Joomla.php

```diff
@@ -491,8 +506,15 @@ public function loadUser($username, $password = NULL) {
       return FALSE;
     }
     $contactID = CRM_Core_BAO_UFMatch::getContactId($uid);
+    $factoryClassName = self::factoryClassName();
     if (!empty($password)) {
-      $instance = JFactory::getApplication('site');
+      if (version_compare(JVERSION, '4.0.0', 'ge')) {
+        $this->loadJoomlaApplication('site');
```

Comment:
I don't think it's right to load the application again here (would have been previously done as part of `loadBootstrap()`). If it's the same as loaded previously, we can just read back `$factoryClassName::$application`. Alternatively, we can use the `$container->get(...)` method to retrieve the required application object without re-invoking parts of the bootstrap sequence.